File: src/main/scala/com/github/mauricio/postgresql/column/BooleanEncoderDecoder.scala
package com.github.mauricio.postgresql.column


/**
 * User: Maurício Linhares
 * Date: 3/5/12
 * Time: 9:50 AM
 */

object BooleanEncoderDecoder extends ColumnEncoderDecoder {

  override def decode(value: String): Any = {
    if ( "t" == value ) {
      true
    } else {
      false
    }
  }

  override def encode( value : Any ) : String = {
    val result = value.asInstanceOf[Boolean]

    if (result) {
      "t"
    } else {
      "f"
    }
  }

}

File: postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/column/BooleanEncoderDecoder.kt
package com.github.jasync.sql.db.postgresql.column

import com.github.jasync.sql.db.column.ColumnEncoderDecoder

object BooleanEncoderDecoder : ColumnEncoderDecoder {

    override fun decode(value: String): Boolean = "t" == value

    override fun encode(value: Any): String {
        val result = value as Boolean
        return if (result) {
            "t"
        } else {
            "f"
        }
    }
}
